primitives/src/pcs/mod.rs

```diff
@@ -192,8 +206,32 @@ pub trait StructuredReferenceString: Sized {
     ///
     /// WARNING: THIS FUNCTION IS FOR TESTING PURPOSE ONLY.
     /// THE OUTPUT SRS SHOULD NOT BE USED IN PRODUCTION.
+    #[cfg(any(test, feature = "test-srs"))]
     fn gen_srs_for_testing<R: RngCore + CryptoRng>(
         rng: &mut R,
         supported_size: usize,
     ) -> Result<Self, PCSError>;
+
+    /// Load public parameter in production environment.
+    /// These parameters are loaded from files with serialized `pp` bytes, and
+    /// the actual setup is usually carried out via MPC and should be
+    /// implemented else where. We only load them into memory here.
+    ///
+    /// If `file=None`, we load the default choice of SRS.
+    fn load_srs_from_file(_supported_size: usize, _file: Option<&Path>) -> Result<Self, PCSError> {
+        unimplemented!("TODO: implement loading SRS from files");
+    }
+}
+
+/// Super-trait specific for univariate polynomial commitment schemes.
+pub trait UnivariatePCS: PolynomialCommitmentScheme {
```

Comment:
```suggestion
pub trait UnivariatePCS: PolynomialCommitmentScheme<Point = <Self as PolynomialCommitmentScheme>::Evaluation> {
```
Because this trait restricts attention to *univariate* polynomials, we know the type of `Point`. We can save the caller the trouble of specifying this bound on `Point` by doing it here ourselves. Asking for a friend. 😉